src/Microsoft.ML.Data/StaticPipe/Estimator.cs

```diff
@@ -69,7 +69,6 @@ string NameMap(PipelineColumn col)
 
                 var readerEst = StaticPipeUtils.GeneralFunctionAnalyzer(Env, ch, input, fakeReconciler, mapper, out var estTail, NameMap);
                 ch.Assert(readerEst == null);
-                ch.AssertValue(estTail);
```

Comment:
>ch.AssertValue(estTail); [](start = 16, length = 24)

I think the correct fix is NOT to allow `nulls` through all of our estimators' `Append` methods, but to have an `if` statement here to only append new estimator if `estTail` is not `null`

Reply:
how about `var est = estTail == null ? AsDynamic : AsDynamic:Append(estTail)` ?

---
In reply to: [229018643](https://github.com/dotnet/machinelearning/pull/1426#discussion_r229018643) [](ancestors = 229018643)